### PR TITLE
Make random seed be int only

### PR DESCRIPTION
**Pull request by oyvindeide**
This is likely to have some user impact, but cant see how the extra complexity is worth it for the users, as seed in most other applications are int.

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by sondreso — COMMENTED** [review]
Couple of questions: 

- Will the users that are using integers today produce the same random numbers after this change? 
- For the users that are using string, can we get a understandable error message, **with** the integer that produces the same random numbers as the string they are currently using?

**Comment by oyvindeide** [comment]
> Cuople of questions:
> 
>     * Will the users that are using integers today produce the same random numbers after this change?
> 
Yes, no change for them, and thus, the majority of users.

>     * For the users that are using string, can we get a understandable error message, **with** the integer that produces the same random numbers as the string they are currently using?

No, not really. The problem is that the current implementation will create a seed sequence if you give a str, where the entropy is a list of ints, unlike when int is used as a seedsequence:

```python
np.random.SeedSequence([1,2,3]).entropy
>>> [1, 2, 3]
```
we have changed this in the past, and they got a new draw. Also, I am not really sure anyone uses this functionality at all.

File: src/ert/enkf_main.py [hunk 1]
```diff
@@ -145,14 +144,11 @@ def _seed_sequence(seed: Optional[str]) -> SeedSequence:
             "add the following random seed to your config file:"
         )
         logger.info(f"RANDOM_SEED {sequence.entropy}")
-        return sequence
-
-    int_seed: Optional[Union[int, Sequence[int]]] = None
-    try:
+        int_seed = sequence.entropy
+    else:
         int_seed = int(seed)
```
Comment (sondreso) [inline src/ert/enkf_main.py:149]: This conversion is not needed anymore 🙂

**Comment by sondreso** [comment]
> Yes, no change for them, and thus, the majority of users.

Great!

> No, not really. The problem is that the current implementation will create a seed sequence if you give a str, where the entropy is a list of ints, unlike when int is used as a seedsequence:
 ```python
np.random.SeedSequence([1,2,3]).entropy
>>> [1, 2, 3]
```
> we have changed this in the past, and they got a new draw. Also, I am not really sure anyone uses this functionality at all.

Right, then I guess the current solution in this PR is as good as any!

**Review by sondreso — APPROVED** [review]

**Pull request merged**